security-analysis/src/test/java/eu/itesla_project/security/SecurityTest.java

```diff
@@ -27,63 +34,144 @@
+    public void setUp() {
         // create pre-contingency results, just one violation on line1
-        Identifiable line1 = Mockito.mock(Identifiable.class);
-        Mockito.when(line1.getId()).thenReturn("line1");
-        LimitViolation line1Violation = new LimitViolation(line1, LimitViolationType.CURRENT, 1000f, "20'", 1100);
-        PreContingencyResult preContingencyResult = new PreContingencyResult(true, Arrays.asList(line1Violation));
+        line1Violation = new LimitViolation("line1", LimitViolationType.CURRENT, 1000f, "20'", 1100);
+        PreContingencyResult preContingencyResult = new PreContingencyResult(true, Arrays.asList(line1Violation), Arrays.asList("action1"));
 
         // create post-contingency results, still the line1 violation plus line2 violation
         Contingency contingency1 = Mockito.mock(Contingency.class);
         Mockito.when(contingency1.getId()).thenReturn("contingency1");
-        Identifiable line2 = Mockito.mock(Identifiable.class);
-        Mockito.when(line2.getId()).thenReturn("line2");
-        LimitViolation line2Violation = new LimitViolation(line2, LimitViolationType.CURRENT, 900f, "10'", 950);
-        PostContingencyResult postContingencyResult = new PostContingencyResult(contingency1, true, Arrays.asList(line1Violation, line2Violation));
-        return new SecurityAnalysisResult(preContingencyResult, Arrays.asList(postContingencyResult));
+        line2Violation = new LimitViolation("line2", LimitViolationType.CURRENT, 900f, "10'", 950);
+        PostContingencyResult postContingencyResult = new PostContingencyResult(contingency1, true, Arrays.asList(line1Violation, line2Violation), Arrays.asList("action2"));
+        result = new SecurityAnalysisResult(preContingencyResult, Arrays.asList(postContingencyResult));
     }
 
     @Test
     public void printPreContingencyViolations() throws Exception {
         StringWriter writer = new StringWriter();
         try {
-            Security.printPreContingencyViolations(createResult(), writer, formatterFactory, formatterConfig, null);
+            Security.printPreContingencyViolations(result, writer, formatterFactory, formatterConfig, null);
         } finally {
             writer.close();
         }
-        assertEquals(writer.toString(), "Pre-contingency violations" + System.lineSeparator() +
-                                               "Action,Equipment,Violation type,Violation name,Value,Limit,Charge %" + System.lineSeparator() +
-                                               ",line1,CURRENT,20',1100.00,1000.0,110" + System.lineSeparator());
+        assertEquals("Pre-contingency violations" + System.lineSeparator() +
+                     "Action,Equipment,Violation type,Violation name,Value,Limit,Charge %" + System.lineSeparator() +
+                     "action1,,,,,," + System.lineSeparator() +
+                     ",line1,CURRENT,20',1100.00,1000.0,110" + System.lineSeparator(),
+                     writer.toString());
```

Comment:
There is a sexiest way to write this kind of statement:
String.join(System.lineSeparator(), a, b, c, d)

Could you write it in a modern way?